### DISCUSSION TITLE
kex_exchange_identification: read: Connection reset by peer

### DISCUSSION THREAD

**Discussion by sahnoun88** [community/community#58249] [New to GitHub] [answered]
I'm trying to connect my github account using ssh  in a fresh installed pop os (ubuntu based distro) , i'm getting this error everytime (for more than a week) : 
kex_exchange_identification: read: Connection reset by peer
Connection reset by 140.82.121.4 port 22

![git_err_disc](https://github.com/community/community/assets/132772009/b3aa845c-fb66-4662-9caa-adc7d2d0b046)
![git_to_disc2](https://github.com/community/community/assets/132772009/2f0c8052-f411-451c-8799-dcfe322125ab)
![ls_ssh](https://github.com/community/community/assets/132772009/3760466d-43de-44ed-bdf8-fd76c13bdde2)

**Comment by aahouzi** [comment]


**Reply by HaniCHERIFyassir** [reply]
this chatgpt answer

**Reply by V3ntus** [reply]
Losers using chatgpt again instead of being helpful

**Comment by sahnoun88** [comment]
all done but the error still appears there .

**Comment by Oualitsen** [comment]
Facing the same issue. does any one have a solution?

**Comment by airtower-luna** [comment, marked as answer]
It looks like *something* is closing your network connection before the SSH handshake happens (or just as it starts). I've seen this behavior from some routers/firewalls trying to block the connection, but with just that information it's hard to be sure. :thinking: 

Does the connection work from another network, or using a VPN?

You could also try [using SSH over the HTTPS port](https://docs.github.com/en/authentication/troubleshooting-ssh/using-ssh-over-the-https-port), sometimes that gets around such filters (if they're not particularly cleverly made). :sweat_smile:

**Reply by cold-bin** [reply]
thank you, the way can work

**Reply by no-chili** [reply]
thanks

**Comment by manjunath-ZEE** [comment]
Even I am facing the same issue. Can anyone help?

**Comment by laruh** [comment]
Hi, got the same error during fetch one time. But then fetch was successful. Have no idea whats going on, so I just live a note for now.
ps: didnt use vpn.
![image](https://github.com/community/community/assets/52405288/f08dfc33-58ac-4f53-977f-5eafbdef5f70)

**Comment by tiger762** [comment]
For what it's worth, I am getting this same error trying to SSH into a VM now running Ubuntu 22.04. After sifting through probably 100 or more useless answers, I looked in a tshark trace and see that the client leads off with its own version string (in the clear). It is expecting the server to send its version string. It doesn't. It just sends a TCP ACK. The client then closes the connection due to the protocol not being followed. So that is closer to what the problem is. The SSH server on (in my case) Ubuntu 22.04 doesn't send its own software version. It has nothing to do with hosts.allow, hosts.deny, "network instability", "just reboot the server and everything is A-OK", sunspot activity, gamma ray bursts. Something with OpenSSH_8.9p1 Ubuntu-3ubuntu0.6, OpenSSL 3.0.2 15 Mar 2022. If anyone has any serious suggestions, I'm all ears.

**Comment by wtz** [comment]
is there an fixed method???

**Comment by readyoneclothing** [comment]
this happens mostly when you connected with your phone network, and your phone internet package expires and you no longer have internet connection,
